### PR TITLE
Fix issues reported by openscanhub

### PR DESCRIPTION
**Pull request by zdohnal**
[Openscanhub](https://github.com/openscanhub/openscanhub) is public tool for doing static analysis of the code (contains several analyzers like Coverity, SNYK, cppcheck, shellcheck etc.) - I initally ran it on version 1.4.5, but some issues are present in master branch as well. 
[Here](https://github.com/michaelrsweet/pappl/files/14208732/scan-results-imp.txt) are the results of the scan.

Does it look ok to have them fixed?

### CODE REVIEW

File: pappl/job-accessors.c [hunk 1]
```diff
@@ -716,9 +716,10 @@ void
 _papplJobSetState(pappl_job_t  *job,	// I - Job
                   ipp_jstate_t state)	// I - New IPP "job-state" value
 {
+  _papplRWLockWrite(job);
```
Comment (michaelrsweet) [inline pappl/job-accessors.c:719]: "job" could be `NULL`, or at least that is the intent of the following check.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Needs some cleanup.

File: pappl/job-accessors.c (hunk 1 above)
Comment (zdohnal) [reply pappl/job-accessors.c:722]: Should be fixed by the last commit. We will check only for job and for job->state after we acquire the lock.

File: pappl/loc.c [hunk 2]
```diff
@@ -300,10 +300,11 @@ _papplLocPrintf(FILE       *fp,		// I - Output file
   va_list	ap;			// Argument pointer
 
 
+  cupsMutexLock(&loc_mutex);
+
   // Load the default message catalog as needed...
   if (!loc_default.pairs)
   {
-    cupsMutexLock(&loc_mutex);
     if (!loc_default.pairs)
```
[the remark this answers is not in the document]
Comment (zdohnal) [reply pappl/loc.c]: Looks like you've changed the code as well, closing.

File: pappl/job-accessors.c [hunk 3]
```diff
@@ -716,32 +716,39 @@ void
...
-	job->state_reasons |= PAPPL_JREASON_ABORTED_BY_SYSTEM;
-      else if (state == IPP_JSTATE_CANCELED)
-	job->state_reasons |= PAPPL_JREASON_JOB_CANCELED_BY_USER;
-
-      if (job->state_reasons & PAPPL_JREASON_ERRORS_DETECTED)
-        job->state_reasons |= PAPPL_JREASON_JOB_COMPLETED_WITH_ERRORS;
-      if (job->state_reasons & PAPPL_JREASON_WARNINGS_DETECTED)
-        job->state_reasons |= PAPPL_JREASON_JOB_COMPLETED_WITH_WARNINGS;
-    }
+  if (job->state == state)
```
Comment (michaelrsweet) [inline pappl/job-accessors.c:724]: Change to `!=` and put the state change code in a block.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
See comments.

**Comment by michaelrsweet** [comment]
[master e8c9517] Fix some issues discovered by OpenScanHub (Issue #334)

**Pull request closed without merging**